x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlParser.java

```diff
@@ -131,6 +131,10 @@ public void exitFunctionExpression(EsqlBaseParser.FunctionExpressionContext ctx)
     private static final BaseErrorListener ERROR_LISTENER = new BaseErrorListener() {
         // replace entries that start with <comma?><space?>DEV_<space?>
         private final Pattern REPLACE_DEV = Pattern.compile(",*\\s*DEV_\\w+\\s*");
+        private final String BEFORE_REGEX = "([\\s,]*)", AFTER_REGEX = "([\\s,]|$)";
+        private final String BEFORE_REPLACEMENT = "$1'", AFTER_REPLACEMENT = "'$2";
+        private final Pattern REPLACE_LP = Pattern.compile(BEFORE_REGEX + "LP" + AFTER_REGEX);
+        private final Pattern REPLACE_LEFT_BRACKET = Pattern.compile(BEFORE_REGEX + "OPENING_BRACKET" + AFTER_REGEX);
```

Comment:
For symmetry, shouldn't we also replace the token for right parens, and for the closing bracket?

Similarly, shouldn't we have a test for `.. | WHERE a in ("foo", ` and an analogous test for `]`?

Reply:
> For symmetry, shouldn't we also replace the token for right parens, and for the closing bracket?

No because those are not lexed as symbols and are still kept as tokens (I _think_ because they simply pop the stack instead pushing twice to a new mode).

Reply:
Thanks for the explanation!